zeebe/engine/src/test/java/io/camunda/zeebe/engine/state/migration/DbMigratorImplTest.java

```diff
@@ -162,4 +162,26 @@ void shouldSetVersionAfterRunningMigrations() {
     // then -- the version is set
     verify(mockMigrationState).setMigratedByVersion(VersionUtil.getVersion());
   }
+
+  @Test
+  void shouldNotRunMigrationsIfTheSameVersion() {
```

Comment:
❌ I'm missing a test that shows what happens when the snapshot is newer than the current version

Reply:
This test should have been added in https://github.com/camunda/zeebe/pull/16421 then, the behavior did not change with this PR.

Reply:
No tests were added that show that these lines throw exceptions

https://github.com/camunda/zeebe/blob/d1c3915ae4bba2f60d10ca611f1670449f519633/zeebe/engine/src/main/java/io/camunda/zeebe/engine/state/migration/DbMigratorImpl.java#L104-L109

Reply:
I'll add a follow to https://github.com/camunda/zeebe/issues/15885 for this then, I think this shouldn't block this PR because it's not new behavior.